src/infra/tokenizer/token-manager.ts

```diff
@@ -0,0 +1,52 @@
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+
+import { FiscalCode } from "@pagopa/io-functions-services-sdk/FiscalCode";
+import { config } from "../../app/config";
+import { tokenizerApiClient } from "./client";
+import { SignerId } from "../../signer/signer";
+
+export const getSignerIdFromFiscalCode = (fiscalCode: FiscalCode) =>
+  pipe(
+    config,
+    TE.fromEither,
+    TE.chain((config) =>
+      pipe(
+        tokenizerApiClient,
+        TE.fromEither,
+        TE.chain((client) =>
+          TE.tryCatch(
+            () =>
+              client.searchUsingPOST({
+                api_key: config.tokenizer.tokenizerApiSubscriptionKey,
+                body: { pii: fiscalCode },
+              }),
+            E.toError
+          )
+        )
+      )
+    )
+  );
+
+export const getFiscalCodeFromSignerId = (signerId: SignerId) =>
+  pipe(
+    config,
+    TE.fromEither,
+    TE.chain((config) =>
+      pipe(
+        tokenizerApiClient,
```

Comment:
DRY! 

try to extract common part of this source file

Reply:
I slightly disagree since make this code genera purpose adds too much complexity in the specific case.